### PR TITLE
ESP32 CAM add Automatic Exposure Control option

### PR DESCRIPTION
**Pull request by sveip**
# What does this implement/fix? 

Quick description and explanation of changes

## Types of changes

- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Other

**Related issue or feature (if applicable):** fixes <link to issue>

**Pull request in [esphome-docs](https://github.com/esphome/esphome-docs) with documentation (if applicable):** esphome/esphome-docs#1711

## Test Environment

- [x] ESP32
- [ ] ESP32 IDF
- [ ] ESP8266

## Example entry for `config.yaml`:

```yaml
esp32_camera:
  name: ir_cam
  external_clock:
    pin: GPIO0
    frequency: 20MHz
  i2c_pins:
    sda: GPIO26
    scl: GPIO27
  data_pins: [GPIO5, GPIO18, GPIO19, GPIO21, GPIO36, GPIO39, GPIO34, GPIO35]
  vsync_pin: GPIO25
  href_pin: GPIO23
  pixel_clock_pin: GPIO22
  power_down_pin: GPIO32
  resolution: XGA
  brightness: 2
  jpeg_quality: 10
  max_framerate: 1 fps
  vertical_flip: false
  horizontal_mirror: false
  aec2: true
  ae_level: 2
  aec_value: 500
```

## Checklist:
  - [x] The code change is tested and works locally.
  - [ ] Tests have been added to verify that the new code works (under `tests/` folder).
  
If user exposed functionality or configuration variables are added/changed:
  - [x] Documentation added/updated in [esphome-docs](https://github.com/esphome/esphome-docs).

### CODE REVIEW

**Review by CarlosGS — APPROVED** [review]
Tested it, works well :)

PS @sveip note in _"Pull request in esphome-docs with documentation..."_ you also need to remove the brackets <> to make the correct hyperlink, to get that PR merged automatically

File: esphome/components/esp32_camera/__init__.py [hunk 1]
```diff
@@ -58,6 +58,9 @@
 CONF_JPEG_QUALITY = "jpeg_quality"
 CONF_VERTICAL_FLIP = "vertical_flip"
 CONF_HORIZONTAL_MIRROR = "horizontal_mirror"
+CONF_AEC2 = "aec2"
+CONF_AE_LEVEL = "ae_level"
+CONF_AEC_VALUE = "aec_value"
```
Comment (oxan) [inline esphome/components/esp32_camera/__init__.py:63]: Can we give these config options a bit more descriptive names?

**Review by oxan — CHANGES_REQUESTED** [review]

**Comment by oxan** [comment]
> to get that PR merged automatically

I wish it were automatic :)

File: esphome/components/esp32_camera/__init__.py (hunk 1 above)
Comment (sveip) [reply esphome/components/esp32_camera/__init__.py:63]: Not really, there is no conclusive doc on this, so I kept them as close to the official esp source as possible to make them referable.

**Review by oxan — APPROVED** [review]

**Pull request merged**